src/core/ConfigManager.cpp

```diff
@@ -713,6 +701,55 @@ void ConfigManager::initDevelopmentWorkingDir()
 	}
 }
 
+void ConfigManager::initPortablePaths()
+{
+	QString applicationPath = qApp->applicationDirPath();
+	m_workingDir = applicationPath + "/lmms-workspace/";
+
+	if (!m_isDevelopment)
+	{
+		m_lmmsRcFile = applicationPath + "/.lmmsrc.xml";
+	}
+}
+
+void ConfigManager::initInstalledPaths()
+{
+	m_workingDir = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation) + "/lmms/";
+
+	// Detect < 1.2.0 working directory as a courtesy
+	if (QFileInfo(QDir::home().absolutePath() + "/lmms/projects/").exists())
+	{
+		m_workingDir = QDir::home().absolutePath() + "/lmms/";
+	}
+
+	// Determine RC file location to use if not in development
+	if (!m_isDevelopment)
+	{
+		QString cfgDir = QStandardPaths::standardLocations(QStandardPaths::AppConfigLocation).at(0);
```

Comment:
There are a couple of issues here:
* `standardLocations` isn't guaranteed to return any locations at all, so this may index the result out of bounds.
* We use the same directory both to read and to write the config file. The paths in `standardLocations` aren't guaranteed to be writable: they may include system-wide config locations, for example.
* In the case of XDG directories, relative paths are considered invalid. However, Qt makes no effort to omit these paths from the result.

I think the easiest way to address these is to use the writable location for configuration files instead. (It's possible for the writable location to be an empty string, or a relative path, in which case I'd just fall back to the legacy location.)

If you want to support reading configuration files from all possible locations, you could consider copying any such file found to the writable location first. You could also include the legacy location as one of these possible locations, which would offer a migration mechanism instead of a fallback mechanism.

Reply:
So, should we just fall back to the legacy path if ``standardLocations`` returns an empty or unwritable path?